MastodonSDK/Sources/MastodonUI/View/Container/MediaGridContainerView.swift

```diff
@@ -200,74 +201,96 @@ extension MediaGridContainerView {
-                containerVerticalStackView.addArrangedSubview(horizontalStackView)
-                horizontalStackView.addArrangedSubview(mediaViews[0])
-                
-                let verticalStackView = createStackView(axis: .vertical)
-                horizontalStackView.addArrangedSubview(verticalStackView)
-                verticalStackView.addArrangedSubview(mediaViews[1])
-                verticalStackView.addArrangedSubview(mediaViews[2])
-            case 4:
-                let topHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(topHorizontalStackView)
-                topHorizontalStackView.addArrangedSubview(mediaViews[0])
-                topHorizontalStackView.addArrangedSubview(mediaViews[1])
-                
-                let bottomHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(bottomHorizontalStackView)
-                bottomHorizontalStackView.addArrangedSubview(mediaViews[2])
-                bottomHorizontalStackView.addArrangedSubview(mediaViews[3])
-            case 5...9:
-                let topHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(topHorizontalStackView)
-                topHorizontalStackView.addArrangedSubview(mediaViews[0])
-                topHorizontalStackView.addArrangedSubview(mediaViews[1])
-                topHorizontalStackView.addArrangedSubview(mediaViews[2])
-                
-                func mediaViewOrPlaceholderView(at index: Int) -> UIView {
-                    return index < mediaViews.count ? mediaViews[index] : UIView()
-                }
-                let middleHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(middleHorizontalStackView)
-                middleHorizontalStackView.addArrangedSubview(mediaViews[3])
-                middleHorizontalStackView.addArrangedSubview(mediaViews[4])
-                middleHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 5))
-                
-                if count > 6 {
-                    let bottomHorizontalStackView = createStackView(axis: .horizontal)
-                    containerVerticalStackView.addArrangedSubview(bottomHorizontalStackView)
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 6))
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 7))
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 8))
-                }
-            default:
-                assertionFailure()
-                return
-            }
+			
+			if count<2 || count>maxCount {
+				assertionFailure("unexpected attachment count \(count)")
+				return
```

Comment:
Since `assert` APIs including `assertionFailure` do nothing in prod builds it would be great to have some sensible behavior here. (If truly impossible this should probably be a `fatalError` instead)

Reply:
This is a sanity check — this condition should really not be possible because it's checked elsewhere. `AdaptiveLayout` is used when there's only one attachment, nothing when zero, and attachments beyond max count are discarded. In Java I'd `throw new IllegalStateException(...)` here which would crash the app.